docs/development/design/3-DN-Retrieving_unique_graphs_from_Otel_data/json_data_converter_HOWTO.md

```diff
@@ -0,0 +1,330 @@
+
+2. The key_path identifies the location of the nested key that will identify the value we are searching for.
+
+3. If the `key_value` attribute is provided, that tells us the structure of the data looks like:
+
+```json
+{
+    "key": "http.method",
+    "value": {
+        "Value": {
+            "StringValue": "GET"
+        }
+    }
+}
+```
+as opposed to:
+```json
+{
+    "key": "value"
+}
+```
+
+
+4. Checks if the `key`'s value matches `http.method` in the first path.
+
+3. Retrieves the value from the specified `value_paths`. For the first path it's specified as `value:Value:StringValue`, giving us a value of "GET".
+
+4. Defines the `value_type` as a `string`
+
+5. Results of the second path are concatenated to the first, giving a final output of "GET 200".
+
+Each field in the field_mapping section follows a similar structure, allowing you to precisely define how to extract and transform data from the input JSON to your desired output format.
+
+When creating your configuration, ensure that each field mapping accurately reflects the structure of your input JSON and the requirements of your target schema.
+
+### 3.6 value_type attribute
+Currently the following two value types are supported
+* string
+* unix_nano (represents date in unix nano format eg. 1723544132228102912)
+
+## 4. Examples
+
+### Example 1: Simple span attribute
+
+To extract `trace_id` we specify the following:
+
+```yaml
+job_id:
+    key_paths: [trace_id]
+    value_type: string
+```
+
+Note the fields `key_value` and `value_paths` are not required.
+
+### Example 2: Referencing a header attribute
+
+Given a header mapping of:
+
+```yaml
+header:
+    paths: [resource:attributes] 
+```
+
+We are now able to access the following json that can be referenced within field_mapping:
+```json
+{
+    "resource": {
+                "attributes": [
+                    {
+                        "key": "service.name",
+                        "value": {
+                            "Value": {
+                                "StringValue": "Test App"
+                            }
+                        }
+                    },
+                    {
+                        "key": "service.version",
+                        "value": {
+                            "Value": {
+                                "StringValue": "1.0"
+                            }
+                        }
+                    }
+                ]
+            }
+}
+```
+
+If we wanted `application_name` to reference "Test App 1.0" we would specify the following config, that concatenates "Test App" with "1.0":
+
+```yaml
+application_name:
+    key_paths: [HEADER:resource:attributes::key, HEADER:resource:attributes::key]
+    key_value: [service.name, service.version]
+    value_paths: [value:Value:StringValue, value:Value:StringValue]
+    value_type: string
+```
+
+### Example 3: Complex field mapping
+
+For more complex scenarios, like concatenating multiple values:
```

Comment:
Also this sentence doesn't really give any context to what follows. Should be something like "An example of a more complex scenario of concatenating multiple values is show below"

However also notice that above an example of concatenating multiple values is shown directly above so is this needed?